src/utils.ts

```diff
@@ -0,0 +1,64 @@
+export class HeaderManager {
```

Comment:
Can we use the [Headers API](https://developer.mozilla.org/en-US/docs/Web/API/Headers) instead? Should be available on both web and Node

Reply:
Great idea though and we should move to using that vs. plain objects/`Record`s

Reply:
That's quite the refactor, it's scoped there: https://github.com/supabase/postgrest-js/pull/619/commits/fe3e80f8cd69224da36dd2094284cbf9c1e7dc98

Reply:
This also require to bump minimal node version to 18 see: [549abdf](https://github.com/supabase/postgrest-js/pull/619/commits/549abdfc0758f6d52cf90652c7b69ce83b8755b3)

LMK if we're comfortable doing that.

Reply:
Ah 😂 I meant we could refactor that in a future PR. Thanks for getting that done!